File: frontend/src/types/summoner.ts
export type Region = 'JP' | 'KR' | 'NA' | 'EUW' | 'EUNE' | 'OCE';

export const REGION_DEFAULT_TAGS: Record<Region, string> = {
  JP: 'JP1',
  KR: 'KR1',
  NA: 'NA1',
  EUW: 'EUW',
  EUNE: 'EUNE',
  OCE: 'OCE1',
};

export interface SuggestCandidate {
  name: string;
  tagLine: string;
  region: Region;
}

export interface RankData {
  queueType: string; // 例: "ランクソロ"
  tier: string; // 例: "CHALLENGER"
  rank: string; // 例: "I"
  leaguePoints: number;
  wins: number;
  losses: number;
}

export interface SummonerData {
  name: string;
  tagLine: string;
  level: number;
  profileIconId: number;
  rank: RankData;
  previousSeasonRank: string;
}

export interface MatchData {
  matchId: string;
  isWin: boolean;
  gameMode: string;
  championName: string;
  kills: number;
  deaths: number;
  assists: number;
  cs: number;
  gameDurationSeconds: number;
  itemIds: number[];
  timeAgoSeconds: number;
}

export interface ChampionStatData {
  championName: string;
  wins: number;
  losses: number;
  cs: number;
  kda: number;
}

File: backend/schemas/summoner.py
from pydantic import BaseModel


class RankData(BaseModel):
    queueType: str
    tier: str  # "UNRANKED" | "IRON" | ... | "CHALLENGER"
    rank: str  # "I" | "II" | "III" | "IV" | ""
    leaguePoints: int
    wins: int
    losses: int


class SummonerData(BaseModel):
    name: str
    tagLine: str
    level: int
    profileIconId: int
    rank: RankData


class RankPositionData(BaseModel):
    position: int  # 同ティア内順位（0=不明）
    totalPlayers: int  # 同ティア内プレイヤー数（0=不明）
    topPercent: float  # 全体上位%


class MatchData(BaseModel):
    matchId: str
    isWin: bool
    gameMode: str
    championName: str
    kills: int
    deaths: int
    assists: int
    cs: int
    gameDurationSeconds: int
    itemIds: list[int]
    timeAgoSeconds: int


class ChampionStatData(BaseModel):
    championName: str
    wins: int
    losses: int
    cs: float
    kda: float
